src/components/Header/Header.js

```diff
@@ -0,0 +1,30 @@
+import { useContext } from 'react';
+import { NavLink } from 'react-router-dom';
+import { UserContext } from '../../context/UserContext';
+import './Header.css';
+
+
+function Header() {
+
+  const { user, setUser } = useContext(UserContext);
+
+  const handleLogout = async () => {
+    setUser(null);
```

Comment:
This sets your user state to null, but it doesn't kill the actual session stored by supabase. Before setting your user to null you should also be calling your `signOut()` function from the auth.js service.